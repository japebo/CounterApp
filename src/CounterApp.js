import React, {useState} from 'react';
import PropTypes from 'prop-types';

const CounterApp = ( {value} ) => {
    
    const [counter, setCounter] = useState(value); //esto es un Hook de React
    
    const handleAdd = () => {
        setCounter( counter + 1 );
    } ;

    const handleSubstract = () => {
        setCounter ( counter - 1 );
    };

    const handleReset = () => {
        setCounter (value);
    };

    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>
            <button onClick = { handleAdd } >+1</button>
            <button onClick = { handleSubstract }>-1</button>
            <button onClick = { handleReset }>Reset</button>
        </>
    );
};

CounterApp.propTypes = {
    value: PropTypes.number
};
CounterApp.defaultProps = {
    value: 1
};

export default CounterApp;