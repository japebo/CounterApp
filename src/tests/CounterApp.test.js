import React from 'react';
import { shallow } from "enzyme";
import CounterApp from "../CounterApp";

describe('Pruebas en <CounterApp />', () => {
    // let wrapper; //de esta forma el inconveniente que tiene es q al ser undefined, visual studio no reconocería sus atributos y métodos propios. Un workaround es el de la próxima línea.
    let wrapper = shallow(<CounterApp />);

    beforeEach( () => {

        wrapper = shallow(<CounterApp />); //se declara de forma global a todas las pruebas para no declararlo dentro de cada prueba una y otra vez.

    });

    test('Debe mostrar <CounterApp /> correctamente (hacer match con el snapshot) y sus valores por defecto', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('Debe mostrar el valor por defecto de 100', () => {
        const wrapper = shallow(<CounterApp value={100}/>);
        
        const h2Text = wrapper.find('h2').text().trim(); //el método trim() se utiliza para quitar los espacios en blanco antes y después del valor del contador (estos espacios fueron definidos en el HTML dentro del Fragment)

        expect(h2Text).toBe('100');
    })
    
    test('Debe incrementar con el botón +1', () => {
        
        // const btn1 = wrapper.find('button').at(0);
        // console.log(btn1.html());

        wrapper.find('button').at(0).simulate('click');
        const h2Text = wrapper.find('h2').text().trim(); //comprobando que el valor se incrmentó en 1

        expect(h2Text).toBe('6');
    })

    test('Debe decrementar con el botón -1', () => {
        
        // const btn1 = wrapper.find('button').at(0);
        // console.log(btn1.html());

        wrapper.find('button').at(1).simulate('click');
        const h2Text = wrapper.find('h2').text().trim(); //comprobando que el valor se incrmentó en 1

        expect(h2Text).toBe('4');
    })
    
    test('Debe colocar el valor por defecto con el botón reset', () => {
        const wrapper = shallow(<CounterApp value={105}/>);

        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click'); //simulando la opresión del botón +1 par de veces para que luego se note el efecto del botón de reset.
        wrapper.find('button').at(2).simulate('click'); //el botón en la posición 2 es el botón de reset.
        
        const h2Text = wrapper.find('h2').text().trim(); 
        // console.log(h2Text);

        expect(h2Text).toBe('105');
    })
    
})
