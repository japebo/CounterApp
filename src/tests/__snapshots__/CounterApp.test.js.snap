// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> Debe mostrar <CounterApp /> correctamente (hacer match con el snapshot) y sus valores por defecto 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    5
     
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</Fragment>
`;
